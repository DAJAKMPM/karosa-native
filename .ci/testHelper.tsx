import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
