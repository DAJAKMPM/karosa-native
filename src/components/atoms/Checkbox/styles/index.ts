/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const CheckboxStyles = StyleSheet.create({
  text: {
    ...theme.textRegular,
    fontWeight: "400",
    marginLeft: 8,
  },
  container: {
    backgroundColor: theme.colors.transparent,
    borderWidth: 0,
    margin: 0,
    padding: 0,
  },
});

export default CheckboxStyles;
