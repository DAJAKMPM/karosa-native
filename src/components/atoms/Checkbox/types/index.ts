/* eslint-disable @typescript-eslint/no-explicit-any */
/**
 *
 * @format
 *
 */

import { ReactElement } from "react";

export type PropsType = {
  value: boolean;
  label: string | ReactElement<any, any>;
  uncheckedColor?: string;
  onPress(): void;
};
