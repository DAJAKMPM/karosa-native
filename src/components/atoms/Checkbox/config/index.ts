/**
 *
 * @format
 *
 */

import { theme } from "@app/styles";

export const defaultColor = theme.colors.primary;
