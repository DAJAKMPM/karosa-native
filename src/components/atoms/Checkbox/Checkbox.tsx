/**
 *
 * Checkbox
 * @format
 *
 */

import React, { FC } from "react";
import { CheckBox as RnCheckbox } from "react-native-elements";

import type { PropsType } from "./types";
import { defaultColor } from "./config";
import CheckboxStyles from "./styles";

const Checkbox: FC<PropsType> = (props) => {
  const { value, label, onPress } = props;

  return (
    <RnCheckbox
      checked={value}
      title={label}
      onPress={onPress}
      checkedColor={defaultColor}
      titleProps={{ style: CheckboxStyles.text }}
      containerStyle={CheckboxStyles.container}
    />
  );
};

export default Checkbox;
