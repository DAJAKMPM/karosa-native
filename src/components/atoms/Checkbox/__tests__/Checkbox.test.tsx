import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import { noop } from "lodash";
import Checkbox from "../Checkbox";

describe("Checkbox Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Checkbox value={false} label="test" onPress={noop} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
