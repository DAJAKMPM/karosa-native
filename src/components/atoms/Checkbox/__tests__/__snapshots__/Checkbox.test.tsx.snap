// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
    testID="checkbox"
  >
    <View>
      <SvgMock
        height={24}
        width={24}
      />
    </View>
    <Text
      style={
        Object {
          "color": "#242424",
          "fontFamily": "SFProText-Regular",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 18.75,
          "marginLeft": 8,
        }
      }
      testID="checkboxTitle"
    >
      test
    </Text>
  </View>
</View>
`;
